lib/app_artifacts.rb

```diff
@@ -27,9 +27,10 @@ def build
 
     # @param [Symbol] name
     # @param [String] path
-    def add_artifact(name, path)
+    # @param [Boolean] allow_missing
+    def add_artifact(name, path, allow_missing: false)
       value = read_artifact(path)
-      raise MissingArtifactError.new("missing artifact: #{path}") if value.nil?
+      raise MissingArtifactError.new("missing artifact: #{path}") if value.nil? && !allow_missing
       value = yield(value) if block_given?
```

Comment:
to go with the "don't yield for nil" comment above
```suggestion
      value = yield(value) if block_given? && value
```